### PR TITLE
deps: use `esbuild-plugins-node-modules-polyfill`

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
This PR replaces `@esbuild-plugins/node-modules-polyfill` with the up-to-date & maintained `esbuild-plugins-node-modules-polyfill`

The `esbuild-plugins` repo itself points towards using `esbuild-plugin-polyfill-node` instead
https://github.com/remorses/esbuild-plugins/blob/373b44902ad3e669f7359c857de09a930ce1ce90/README.md?plain=1#L15-L16

After doing this in the Remix repo (see https://github.com/remix-run/remix/pull/5274), we got quite some new bugs, so we chose to go for @imranbarbhuiya's  `esbuild-plugins-node-modules-polyfill` instead (see https://github.com/remix-run/remix/pull/6562), which is an up-to-date and well maintained alternative

Added benefit is that we won't get the following deprecation warnings when installing `@esbuild-plugins/node-modules-polyfill`:

```sh
npm WARN deprecated rollup-plugin-inject@3.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-inject.
npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
```

### CODE REVIEW

**Review by adamraine — COMMENTED** [review]
Interesting, let's see if our CI jobs work. @connorjclark any thoughts?

File: build/build-bundle.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ import {createRequire} from 'module';
 
 import esMain from 'es-main';
 import esbuild from 'esbuild';
+import {nodeModulesPolyfillPlugin} from 'esbuild-plugins-node-modules-polyfill';
```
[the remark this answers is not in the document]
Comment (MichaelDeBoey) [reply build/build-bundle.js:19]: It should be a drop in replacement normally 🤔

Added this line to `esbuild-plugins.js` as well as requested

**Comment by connorjclark** [comment]
The failing tests show two critical issues:

1) zlib inflate code is not being removed. Maybe the zlib polyfil is no longer split into `__zlib-lib/inflate` and `__zlib-lib/deflate` modules, which would explain that. This is all CI shows.
2) more importantly, the bundled smoke tests fail totally. Comment out that zlib check throwing in CI, and then run `DEBUG=1 yarn build-devtools` and then `yarn test-bundle seo` you'll see:

```
Error: Worker returned an error: ReferenceError: navigator is not defined
Log:
[STDERR] ReferenceError: navigator is not defined
    at node-modules-polyfills:process (file:///Users/cjamcl/src/lighthouse/dist/node-modules-polyfills:process:63:9)
    at __init (file:///Users/cjamcl/src/lighthouse/dist/lighthouse-dt-bundle.js:20:58)
    at build/process-global.js (file:///Users/cjamcl/src/lighthouse/build/process-global.js:1:1)
    at __init (file:///Users/cjamcl/src/lighthouse/dist/lighthouse-dt-bundle.js:20:58)
    at file:///Users/cjamcl/src/lighthouse/clients/devtools/devtools-entry.js:1:1
    at file:///Users/cjamcl/src/lighthouse/dist/lighthouse-dt-bundle.js:93839:3
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
```

This error seems very non-trivial, and I'm not sure where to begin to fix it. The current setup works with no issues for us so I don't think there's any urgent motivation for us to change right now.

**Comment by MichaelDeBoey** [comment]
I think @imranbarbhuiya can probably help out with fixing all these errors

**Comment by adamraine** [comment]
Please let us know if get this to work, as we are interested in an alternative solution. For now we will close this PR

**Comment by imranbarbhuiya** [comment]
@adamraine I checked the file https://github.com/GoogleChrome/lighthouse/blob/f33cf9375583c384efa4379d55df1f1936406d2b/build/build-bundle.js#L111 and saw you are adding some empty functions for `__zlib/inflate` so can I know why it's needed? Are you using this polyfill anywhere in your code? If yes, why changing it to empty export? Does the `empty` option in `modules` fixes the issue?

**Comment by connorjclark** [comment]
The intention is to exclude the large amount of code needed to decompress data. Our library only needs to compress.

**Pull request closed without merging**